### PR TITLE
CLDR 42 Re-Update - Country with data sorting

### PR DESCRIPTION
**Pull request by gouniLee**
### Checklist

* [ ] At least one test case is included for this feature or bug fix.
* [ ] `ReleaseNotes` has added or is not needed.
* [ ] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [ ] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

### Resolution
I found the `regionnames.json` file were not update when I was working cldr v42 update. and the tool code (`gencountrynames.js`) haven't been sorting data by using 'json-stable-stringify`. so I updated tool code and update output agagin base on v42. 
then Update cldr v43 work will do again based on this change.

### Links
[//]: # (Related issues, references)

### CODE REVIEW

**Comment by gouniLee** [comment]
@ehoogerbeets Please check my PRs regarding cldr update.

File: js/data/locale/af/NA/ctrynames.json
Comment (ehoogerbeets) [inline js/data/locale/af/NA/ctrynames.json:221]: The difference for most of these files is that it is sorting differently.... This should not happen.

File: tools/cldr/gencountrynames.js
Comment (ehoogerbeets) [inline tools/cldr/gencountrynames.js:322]:
```
fs.writeFileSync(path.join(pathname, filePrefix + "reverse.json"), stringify(reverse, {
    space: 4,
    cmp: function(a, b) {
        return obj[a] < obj[b] ? -1 : obj[a] > obj[b] ? 1 : 0;
    }
}), "utf-8");
```

File: js/data/locale/af/NA/ctrynames.json
Comment (ehoogerbeets) [reply js/data/locale/af/NA/ctrynames.json:221]: The difference is that json-stable-stringify uses a different comparison function. It was already sorted before, so we just need to pass in the comparison function to get it to sort the same way it did before.

**Review by ehoogerbeets — APPROVED** [review]

**Review by ehoogerbeets — COMMENTED** [review]
Update the sorting

**Comment by ehoogerbeets** [comment]
Accidentally approved it. Please update the sorting before merging.

File: js/data/locale/af/NA/ctrynames.json
Comment (gouniLee) [reply js/data/locale/af/NA/ctrynames.json:221]: When you see the `regionnames.json` file that is generated with the tool code as well, I can see the data is not sorted.
i.e.
current data.
```
"AD" : [
  "name":"Canillo",
  "code": "02"
]
```
I expect to get a result like below.
```
"AD" : [
  "code": "02"
  "name":"Canillo",
]
```

**Comment by gouniLee** [comment]
I updated it.

File: js/data/locale/af/regionnames.json
Comment (ehoogerbeets) [inline js/data/locale/af/regionnames.json:4]: Why is this one not sorted properly? "code" should come before "name" ...

Comment (gouniLee) [reply js/data/locale/af/regionnames.json]: yeah...I re-updated the tool code and generated `regionnames.json` files again to have the correct sorted data. Please take a look them again

**Comment by gouniLee** [comment]
I am going to merge this PR. All of the data is sorted and got approval before.

**Pull request merged**
